src/webgpu/shader/execution/expression/binary/af_data.ts

```diff
@@ -0,0 +1,286 @@
+  { lhs: 10.0, rhs: kValue.f64.negative.min, expected: [kValue.f64.negative.min] },
+  { lhs: 10.0, rhs: -10.0, expected: [0.0] },
+  { lhs: 10.0, rhs: -1.0, expected: [9] },
+  { lhs: 10.0, rhs: -0.125, expected: [9.875] },
+  { lhs: 10.0, rhs: kValue.f64.negative.max, expected: [10.0] },
+  { lhs: 10.0, rhs: kValue.f64.negative.subnormal.min, expected: [0.0, 10.0] },
+  { lhs: 10.0, rhs: kValue.f64.negative.subnormal.max, expected: [0.0, 10.0] },
+  { lhs: 10.0, rhs: 0.0, expected: [10.0] },
+  { lhs: 10.0, rhs: kValue.f64.positive.subnormal.min, expected: [0.0, 10.0] },
+  { lhs: 10.0, rhs: kValue.f64.positive.subnormal.max, expected: [0.0, 10.0] },
+  { lhs: 10.0, rhs: kValue.f64.positive.min, expected: [10.0] },
+  { lhs: 10.0, rhs: 0.125, expected: [10.125] },
+  { lhs: 10.0, rhs: 1.0, expected: [11] },
+  { lhs: 10.0, rhs: 10.0, expected: [20] },
+  { lhs: 10.0, rhs: kValue.f64.positive.max, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: kValue.f64.negative.min, expected: [0.0] },
+  { lhs: kValue.f64.positive.max, rhs: -10.0, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: -1.0, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: -0.125, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: kValue.f64.negative.max, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: kValue.f64.negative.subnormal.min, expected: [0.0, kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: kValue.f64.negative.subnormal.max, expected: [0.0, kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: 0.0, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: kValue.f64.positive.subnormal.min, expected: [0.0, kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: kValue.f64.positive.subnormal.max, expected: [0.0, kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: kValue.f64.positive.min, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: 0.125, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: 1.0, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: 10.0, expected: [kValue.f64.positive.max] },
+  { lhs: kValue.f64.positive.max, rhs: kValue.f64.positive.max, expected: [kValue.f64.positive.max, kValue.f64.positive.infinity] }
+
+];
+
+/** Hashing function for keys in lookup tables */
+function hashRawDataInputs(lhs: number, rhs: number): number {
+  return crc32(`lhs: ${lhs}, rhs: ${rhs}`);
+}
+
+/** Table mapping addition binary inputs to expectation intervals */
+const addition_table: Record<number, FPInterval> = {};
+addition_raw_values.forEach(value => {
+  addition_table[hashRawDataInputs(value.lhs, value.rhs)] = FP.abstract.spanIntervals(
+    ...value.expected.map(e => FP.abstract.correctlyRoundedInterval(e))
+  );
+});
```

Comment:
nit: Prefer `for..of`, but better, construct all at once so the result can be readonly:

```suggestion
const kAdditionTable: Readonly<Record<number, FPInterval>> = Object.fromEntries(
  kAdditionRawValues.map(value => [
    hashRawDataInputs(value.lhs, value.rhs),
    FP.abstract.spanIntervals(...value.expected.map(e => FP.abstract.correctlyRoundedInterval(e))),
  ])
);
```

(this should work, but please make sure)